packages/fetchai/skills/generic_seller/behaviours.py

```diff
@@ -139,30 +139,63 @@ def _register_agent(self) -> None:
         self.context.outbox.put_message(message=oef_search_msg)
         self.context.logger.info("registering agent on SOEF.")
 
-    def register_service_personality_classification(self) -> None:
+    def register_service(self) -> None:
         """
-        Register the agent's service, personality and classification.
+        Register the agent's service.
 
         :return: None
         """
         strategy = cast(GenericStrategy, self.context.strategy)
-        descriptions = [
-            strategy.get_register_service_description(),
-            strategy.get_register_personality_description(),
-            strategy.get_register_classification_description(),
-        ]
+        description = (strategy.get_register_service_description(),)
         oef_search_dialogues = cast(
             OefSearchDialogues, self.context.oef_search_dialogues
         )
-        for description in descriptions:
-            oef_search_msg, _ = oef_search_dialogues.create(
-                counterparty=self.context.search_service_address,
-                performative=OefSearchMessage.Performative.REGISTER_SERVICE,
-                service_description=description,
-            )
-            self.context.outbox.put_message(message=oef_search_msg)
+        oef_search_msg, _ = oef_search_dialogues.create(
+            counterparty=self.context.search_service_address,
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            service_description=description,
+        )
+        self.context.outbox.put_message(message=oef_search_msg)
         self.context.logger.info("registering service on SOEF.")
 
+    def register_genus(self) -> None:
```

Comment:
Re almost identical code blocks..why not extract common functionality in a private method so it only exists once?